universal.c

```diff
@@ -512,13 +512,11 @@ XS(XS_UNIVERSAL_can)
             GV * const gv = gv_fetchmethod_sv_flags(pkg, ST(i), 0);
             if (!gv || !isGV(gv) || !GvCV(gv))
                 XSRETURN_UNDEF;
+            else {
+                ST(i-1) = sv_2mortal(newRV(MUTABLE_SV(GvCV(gv))));
+            }
         }
-
         EXTEND(SP, items - 1);
```

Comment:
You moved the writes to the stack before the EXTEND(), which may write beyond the end of the stack, the EXTEND should be moved above the loop.